zerver/views/auth.py

```diff
@@ -242,7 +246,7 @@ def maybe_send_to_registration(
         {"email": email},
         realm=realm,
         from_multiuse_invite=from_multiuse_invite,
-        invited_as=invited_as,
+        invited_as=role or invited_as,
```

Comment:
We probably need do document in the `docs/production/authentication-methods.md` docs that `role` in SAML takes precedence over a `role` specified in an invitation, if a user is created through a SAML login after an invitation. I agree that's correct behavior, but maybe we should comment on it.

> # role provided by the authentication backend takes precedence over an invitation, because it is likely being synced from the authentication database.

Reply:
Not sure where best to add it, for now I did it on top of the SAML section of the doc:

![image](https://github.com/user-attachments/assets/616e4726-dba1-4c8a-92e1-a113f86b5c07)

Reply:
adding it on top of the entire `docs/production/authentication-methods.md` seems like maybe giving it too much importance when it's only SAML that supports it, so might just act to distract readers in 95% of cases (which don't use SAML)?

Reply:
Yeah I think we should just put it in the subsection on syncing data for SAML.

Reply:
Moved this in my push just now.